AsyncDisplayKit/ASPagerFlowLayout.m

```diff
@@ -0,0 +1,63 @@
+//
+//  ASPagerFlowLayout.m
+//  AsyncDisplayKit
+//
+//  Created by Levi McCallum on 2/12/16.
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import "ASPagerFlowLayout.h"
+
+@interface ASPagerFlowLayout ()
+
+@property (strong, nonatomic) NSIndexPath *currentIndexPath;
+
+@end
+
+@implementation ASPagerFlowLayout
+
+- (void)invalidateLayout
+{
+  self.currentIndexPath = [self _indexPathForVisiblyCenteredItem];
```

Comment:
It's basically called whenever there's a bounds change. Technically the `currentIndexPath` should be the current index path even after the invalidation, as we're keeping state here of what item was centered before the bounds change and ensuring that it remains centered after the layout rebuilds itself.